node/node.go

```diff
@@ -558,6 +571,51 @@ func createPEXReactorAndAddToSwitch(addrBook pex.AddrBook, config *cfg.Config,
 	return pexReactor
 }
 
+// startStateSync starts an asynchronous state sync process, then switches to fast sync mode.
+func startStateSync(ssR *statesync.Reactor, bcR fastSyncReactor, conR *consensus.Reactor,
+	config *cfg.StateSyncConfig, fastSync bool, stateDB dbm.DB, blockStore *store.BlockStore) error {
+	ssR.Logger.Info("Starting state sync")
+
+	state := sm.LoadState(stateDB)
+	stateSource, err := statesync.NewLightClientStateSource(state.ChainID, state.Version,
+		config.RPCServers, lite.TrustOptions{
+			Period: config.TrustPeriod,
+			Height: config.TrustHeight,
+			Hash:   config.TrustHashBytes(),
+		}, ssR.Logger.With("module", "lite"))
+	if err != nil {
+		return fmt.Errorf("failed to set up light client: %w", err)
+	}
+
+	go func() {
+		state, commit, err := ssR.Sync(stateSource)
+		if err != nil {
+			ssR.Logger.Error("State sync failed", "err", err)
+			return
+		}
+		err = sm.BootstrapState(stateDB, state)
+		if err != nil {
+			ssR.Logger.Error("Failed to bootstrap node with new state", "err", err)
+			return
+		}
+		blockStore.SaveSeenCommit(state.LastBlockHeight, commit)
```

Comment:
does this always succeed?

Reply:
Ah, nice catch - it's sort of hard to keep track of which functions return errors and which panic, guess I just assumed this paniced.